test/unit/files/test_posix.py

```diff
@@ -333,6 +334,84 @@ def test_posix_user_access_using_boolean_rules():
     _assert_user_access_granted(file_sources, user_context)
 
 
+def test_posix_file_url_mode_non_admin_cannot_retrieve():
+    with tempfile.NamedTemporaryFile(mode="w") as tf:
+        tf.write("File content returned from a file:// location")
+        tf.flush()
+        test_url = f"file://{tf.name}"
+        user_context = user_context_fixture()
+        file_sources = _configured_file_sources(empty_root=True)
+        file_source_pair = file_sources.get_file_source_path(test_url)
+
+        assert file_source_pair.path == test_url.replace("file://", "")
```

Comment:
I guess this can be simplified to:

```suggestion
        assert file_source_pair.path == tf.name
```

Also below.